src/lib/phdi-building-blocks/phdi_building_blocks/conversion.py

```diff
@@ -318,9 +324,25 @@ def convert_message_to_fhir(
             {"name": "rootTemplate", "valueString": root_template},
         ],
     }
-    response = requests.post(
-        url=url, json=data, headers={"Authorization": f"Bearer {access_token}"}
+    access_token = cred_manager.get_access_token().token
+    headers = {"Authorization": f"Bearer {access_token}"}
+
+    retry_strategy = Retry(
+        total=3,
+        status_forcelist=[429, 500, 502, 503, 504],
+        allowed_methods=["HEAD", "POST", "OPTIONS"],
     )
+    adapter = HTTPAdapter(max_retries=retry_strategy)
+    http_session = requests.Session()
+    http_session.mount("https://", adapter)
+    response = http_session.post(url=url, json=data, headers=headers)
+
+    # Retry with new token in case it expired since creation (or from cache)
+    if response.status_code == 401:
+        access_token = cred_manager.get_access_token().token
+        reauth_headers = headers.copy()
+        reauth_headers["Authorization"] = f"Bearer {access_token}"
```

Comment:
This is sort of a nit, but is there any benefit to copying the existing headers object and overwriting the token vs. just creating a new object literal?

Reply:
This has been updated so as not to copy the headers dict, and tests just check the final call to ensure the updated access token was supplied.  

Tests also confirm there were 2 get requests.  I chatted with Bryan on this, and he agrees this is sufficient for unit tests.